peer/brontide_test.go

```diff
@@ -1118,3 +1127,43 @@ func TestPeerCustomMessage(t *testing.T) {
 	require.Equal(t, remoteKey, receivedCustom.peer)
 	require.Equal(t, receivedCustomMsg, &receivedCustom.msg)
 }
+
+// TestPeerUnknownMessageOdd tests custom message exchange between peers.
+func TestPeerUnknownMessageOdd(t *testing.T) {
+	t.Parallel()
+	_, _, _, mockConn := SetUpTest(t)
+
+	// Receive an unknown message.
+	_, err := lnwire.NewUnknown(
+		lnwire.MessageType(32001), []byte{4, 5, 6},
+	)
+	require.NoError(t, err)
+
+	receivedData := []byte{0x7d, 0x01, 0x4, 0x5, 0x6}
+	mockConn.readMessages <- receivedData
+
+	// Todo: would be better not to have to wait
```

Comment:
I don't know how to explicitly wait for brontide to do its thing.. and the connection closes after the test finishes. It seemed 'ok' to wait for the connection to close since time.Sleep is used in a variety of places.

If i know how to properly wait (without involving timer.Sleep) then i would.. that would be the correct thing to do.

Reply:
I think you can use the `WaitForDisconnect` function in both cases. In the unknown even case, the `WaitForDisconnect` function should return within X seconds. In the unknown case, the `WaitForDisconnect` function should not return within X seconds

Reply:
I got stuck on this one. Will revisit.